### PR TITLE
OPIK-472 dataset items CRUD tests

### PR DESCRIPTION
**Pull request by AndreiCautisanu**
## Details

Added tests for CRUD operations for dataset items.
Test coverage:

1. Dataset item insertion - all possible combinations (insert items via the UI/SDK into dataset created via UI/SDK - check data correct and in sync between UI and SDK)
2. Dataset item modification - update dataset items with new data via the SDK and check the change is propagated and visible in both UI and SDK
3. Dataset item deletion - delete dataset item via both the SDK and the UI and check that the removal is propagated to both UI and SDK and that the new dataset is displayed/returned correctly (deleted item is no longer present)
4. Dataset item clearance - mass delete every item in the dataset by using dataset.clear() and check that the dataset appears as empty in both UI and SDK

7 test instances total. Tests 2, 3, and 4 are only done on datasets created via SDK and items inserted via the SDK to save on time (insertion and creation via all other ways already tested in test 1). @ferc let me know if you feel it's necessary to add the combinations from test 1 to the other tests, should be as simple as adding parametrization to all of them.

### CODE REVIEW

**Comment by AndreiCautisanu** [comment]
Ran sanity, datasets, and all_features successfully: https://github.com/comet-ml/opik/actions/workflows/end2end_suites.yml

File: tests_end_to_end/tests/Datasets/datasets_utils.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    dataset_items_page = DatasetItemsPage(page)
+    item = dataset_items_page.delete_first_item_on_page_and_return_content()
+    return item
+
+
+def wait_for_dataset_to_be_visible(client: Opik, dataset_name: str, timeout=10):
+    start_time = time.time()
+    dataset = None
+    while time.time() - start_time < timeout:
+        dataset = None
```
Comment (alexkuzmik) [inline tests_end_to_end/tests/Datasets/datasets_utils.py:73]: Lets put a small sleep ( e.g. 0.1 seconds) after every attempt
Btw we have quite similar utility in `opik.synchronization`

File: tests_end_to_end/tests/Datasets/datasets_utils.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    items = []
+    while time.time() - start_time < timeout:
+        items = dataset.get_items()
+        if len(items) == expected_items_number:
+            expected_number_achieved = True
+            break
+        time.sleep(1)
+
+    if not expected_number_achieved:
+        raise Exception(f'expected to see {expected_items_number} in dataset, instead found {len(items)} after {timeout} seconds of retries')
```
Comment (alexkuzmik) [inline tests_end_to_end/tests/Datasets/datasets_utils.py]: I recommend raising AssertionError instead. That way pytest will mark a test as Failed (F), and not Error (E).
It helps with distinguishing when the problem is in the code and when it's with the test

Comment (AndreiCautisanu) [reply tests_end_to_end/tests/Datasets/datasets_utils.py]: good call, added

File: tests_end_to_end/tests/Datasets/datasets_utils.py (hunk 1 above)
Comment (AndreiCautisanu) [reply tests_end_to_end/tests/Datasets/datasets_utils.py:73]: slipped my mind, added it to the other wait but forgot about it here, thank you for the catch, added

**Review by alexkuzmik — APPROVED** [review]

**Pull request merged**